### ISSUE TITLE
Relation widget crashes if `valueField` doesn't return string or component.

### ISSUE THREAD

**Issue by tech4him1** [netlify/netlify-cms#1116]
**- Do you want to request a *feature* or report a *bug*?**
bug

**- What is the current behavior?**
If the `valueField` for a relation widget isn't a string (or other renderable object in React), the CMS will crash.
**- If the current behavior is a bug, please provide the steps to reproduce.**
Example:
```yaml
    fields:
      - label: "Related Post Date"
        name: "post_date"
        widget: "relation"
        collection: "posts"
        searchFields: ["title", "date"]
        valueField: "date"
```
Try to type into the relation widget.
**- What is the expected behavior?**
The relation widget should still work with fields that don't return strings or React components.

**- Please mention your CMS, node.js, and operating system version.**

1.2.0-dev

**- Please link or paste your `config.yml` below if applicable.**

**Comment by tech4him1** [comment]
This could potentially be "fixed" by converting the values to strings before including them in the relation widget. It would probably need done at: https://github.com/netlify/netlify-cms/blob/952a6d509f4e681d4757bef51d4b2011c72f77ad/src/components/EditorWidgets/Relation/RelationControl.js#L90 and https://github.com/netlify/netlify-cms/blob/952a6d509f4e681d4757bef51d4b2011c72f77ad/src/components/EditorWidgets/Relation/RelationControl.js#L106
I don't see any problem with just converting the value to a string when suggesting options (first code), but I don't like doing that for the actual saved value (second code). The saved value in the entry should be the actual value that's in the related post.

**Comment by tech4him1** [comment]
Ah, looks like the actual saved value is captured separately:
https://github.com/netlify/netlify-cms/blob/952a6d509f4e681d4757bef51d4b2011c72f77ad/src/components/EditorWidgets/Relation/RelationControl.js#L66 https://github.com/netlify/netlify-cms/blob/952a6d509f4e681d4757bef51d4b2011c72f77ad/src/components/EditorWidgets/Relation/RelationControl.js#L84

My simple fix proposed above should work, then.

**Comment by barthc** [comment]
Issue fixed in #1936
